### DISCUSSION TITLE
`BaseSettings` attribute depends on another attribute

### DISCUSSION THREAD

**Discussion by lorenabalan** [pydantic/pydantic#5212] [Question] [answered]
Hi there! I'm becoming a big fan of `BaseSettings` these days and I'm wondering if there's a way to have something like

```python
from pydantic import BaseSettings


class Settings(BaseSettings):
    env: str = "dev"
    my_url: str = f"my-{env}-url"
```
where `my_url` is updated if `env` is updated, i.e.

```python
import os

os.environ["ENV"] = "prod"
settings = Settings()
print(settings.env)  # prod
print(settings.my_url)  # should be `my-prod-url`, but it's `my-dev-url` instead
```

I'd like to avoid setting `my_url` as an environment variable as well. I couldn't find a thread that answers this, but do let me know if there's something I missed!

**Comment by dmontagu** [comment, marked as answer]
It sounds like you should use an `@property` for this — does that have any issues?

**Comment by lorenabalan** [comment]
Ah yesss good point, thank you! 🙌  This works as intended:

```python
class Settings(BaseSettings):
    env: str = "dev"

    @property
    def my_url(self):
        return f"my-{self.env}-url"
```
